src/pybind/mgr/volumes/fs/subvolspec.py

```diff
@@ -86,17 +86,23 @@ def trash_dir(self):
         """
         return os.path.join(self.subvolume_prefix.encode('utf-8'), b"_deleting")
 
-    def make_subvol_snap_path(self, snapdir, snapname):
+    def make_subvol_snap_path(self, snapdir, snapname=None):
         """
-        return the subvolume snapshot path for a given snapshot name
+        return the subvolume snapshot path for a given snapshot name or return the snapdir path
```

Comment:
write a separate method to return the snap dir path and call it, make_subvol_snapdir_path()